source/directives/text.ts

```diff
@@ -0,0 +1,48 @@
+/// <reference path="../../node_modules/forms-js/dist/forms-js.d.ts" />
+
+module adaptor.directives {
+
+  export function TextDirective($log:ng.ILogService):ng.IDirective {
+
+    return {
+      require: '^fjsForm',
+      restrict: 'EA',
+      templateUrl: '/templates/text.html',
+
+      scope: {
+        fieldName: '@'
+      },
+
+      link: function($scope:any,
+                     $element:ng.IAugmentedJQuery,
+                     $attributes:ng.IAttributes,
+                     formForController:any):void {
+
+        if (!$scope.fieldName) {
+          $log.error('Missing required field "fieldName"');
+
+          return;
+        }
+
+        $scope.attributeMetadata = formForController.form.registerAttribute($scope.fieldName);
+        $scope.watchPath = 'attributeMetadata.form.formData.' + $scope.fieldName;
+
+        // TODO This is likely a race condition.
+        // TODO Refactor this to be shared between input directives.
+        $scope.$watch($scope.watchPath, function(value:any) {
+          $scope.bindable = formsjs.Flatten.read($scope.fieldName, $scope.attributeMetadata.form.formData);
```

Comment:
Yes it's hard working with `ng-model` sometimes, that's why I did that weird string replacement in the template function. In schema form we solve  it in a similar way but instead of an template function we do a "manual" `$compile` of the template. 

I'm not sure which is best, doing string replacement to the template or doubling up on the $watches, small watches like that *should* be fast.

Anyways there is a lib a user created for schema form, https://github.com/mike-marcacci/objectpath, that parses a flattened array notation to something `ng-model` can bind directly to, even with strange things in the model name.